mlflow/server/_job_manager.py

```diff
@@ -0,0 +1,183 @@
+import logging
+import threading
+import time
+from concurrent.futures import ThreadPoolExecutor
+from typing import Any
+
+from mlflow.environment_variables import MLFLOW_MAX_CONCURRENT_PROMPT_OPTIMIZATION_JOBS
+from mlflow.exceptions import MlflowException
+from mlflow.genai.optimize.types import LLMParams, OptimizerConfig
+from mlflow.genai.scorers.builtin_scorers import get_builtin_scorer_by_name
+from mlflow.protos.databricks_pb2 import INVALID_PARAMETER_VALUE
+from mlflow.protos.service_pb2 import GetOptimizePromptJob
+
+_logger = logging.getLogger(__name__)
+
+
+# Prompt Optimization Job Management
+class PromptOptimizationJobManager:
+    """Manages prompt optimization jobs using a thread pool for controlled concurrency."""
+
+    def __init__(self):
+        self._jobs: dict[str, dict[str, Any]] = {}
+        self._next_job_id = 0
+        self._lock = threading.Lock()
+        self._executor = ThreadPoolExecutor(
+            max_workers=MLFLOW_MAX_CONCURRENT_PROMPT_OPTIMIZATION_JOBS.get(),
+            thread_name_prefix="prompt_opt",
+        )
+
+    def create_job(
+        self,
+        *,
+        train_dataset_id: str,
+        eval_dataset_id: str | None,
+        prompt_url: str,
+        scorers: list[Any],
+        target_llm: str,
+        algorithm: str | None,
+    ) -> str:
+        """Create a new prompt optimization job."""
+        with self._lock:
+            job_id = str(self._next_job_id)
+            self._next_job_id += 1
+
+            self._jobs[job_id] = {
```

Comment:
Is there a reason why we use dict here?

Reply:
it's for quick querying job by job-id.

Reply:
what do you mean by quick?

Reply:
If there is no reason, we should use dataclass.